File: src/ch1/caesar_enc2.py
# 암호화 함수
def encrypt(text, shift):
    a = ord('A')
    conv = lambda n: chr((ord(n) - a + shift) % 26 + a)
    enc1 = lambda n: conv(n) if 'A' <= n <= 'Z' else n
    return ''.join([enc1(n) for n in text])

# 함수를 실행
enc = encrypt("I LOVE RUST.", 3)
dec = encrypt(enc, -3)
print(enc, "=>", dec)

File: src/ch1/caesar_enc2.rs
// 암호화 함수
fn encrypt(text: &str, shift: i16) -> String {
    let a = 'A' as i16;
    let is_az = |c| 'A' <= c && c <= 'Z';
    let conv = |c| (((c-a+shift+26)%26+a) as u8)as char;
    let enc1 = |c| if is_az(c) { conv(c as i16) } else { c };
    text.chars().map(|c| enc1(c)).collect()
}

fn main() {
    let enc = encrypt("I LOVE RUST.", 3);
    let dec = encrypt(&enc, -3);
    println!("{} => {}", enc, dec);
}